protocol_connect.go

```diff
@@ -154,6 +162,42 @@ func (h *connectHandler) NewConn(
 		}
 	}
 
+	var requestBody io.ReadCloser
+	var contentType, codecName string
+	if request.Method == http.MethodGet {
+		query := request.URL.Query()
+		if failed == nil && !query.Has(connectUnaryEncodingQueryParameter) {
+			failed = errorf(CodeInvalidArgument, "missing %s parameter", connectUnaryEncodingQueryParameter)
+		} else if failed == nil && !query.Has(connectUnaryMessageQueryParameter) {
+			failed = errorf(CodeInvalidArgument, "missing %s parameter", connectUnaryMessageQueryParameter)
+		}
+		msgReader := io.Reader(strings.NewReader(query.Get(connectUnaryMessageQueryParameter)))
```

Comment:
After in-person discussion: I think we can simplify here by getting rid of our custom b64 reader and choosing the correct decoder based on string length mod 4.

Reply:
Fixed in 0054f7c9032ef3f60b5f131b7a7bfe4706996c39.